api/src/services/users.ts

```diff
@@ -40,34 +41,39 @@ export class UsersService extends ItemsService {
 	 * User email has to be unique case-insensitive. This is an additional check to make sure that
 	 * the email is unique regardless of casing
 	 */
-	private async checkUniqueEmails(emails: string[], excludeKey?: PrimaryKey) {
-		if (emails.length > 0) {
-			const query = this.knex
-				.select('email')
-				.from('directus_users')
-				.whereRaw(`LOWER(??) IN (${emails.map(() => '?')})`, ['email', ...emails]);
-
-			if (excludeKey) {
-				query.whereNot('id', excludeKey);
-			}
+	private async checkUniqueEmails(emails: string[], excludeKey?: PrimaryKey): Promise<void> {
+		if (emails.some((email) => email === null)) {
+			throw new ContainsNullValuesException('email', {
```

Comment:
This could use FailedValidationException and then we wouldn't have to deal with "CONTAINS_NULL_VALUES" in the front. But it feels kinda hacky mimicing the Joi error format to get the response we want. What's your opinion @rijkvanzanten?